SimPEG/Optimization.py

```diff
@@ -939,6 +943,72 @@ def approxHinv(self, value):
         self._approxHinv = value
 
     @Utils.timeIt
+    def minimize(self, evalFunction, x0):
```

Comment:
still have to write.
can you point to a test for the other solvers?
Still have to write a test for the compact norms, but I have an idea for that